lib/kvdb/ikvdb.c

```diff
@@ -2494,7 +2498,7 @@ ikvdb_kvs_put(
     vbufsz = tls_vbufsz;
     vbuf = NULL;
 
-    if (clen == 0 && vlen > kk->kk_vcompmin && !(flags & HSE_KVS_PUT_VCOMP_OFF)) {
+    if (clen == 0 && is_compression_allowed(kk, flags) && vlen > kk->kk_vcompmin) {
```

Comment:
We should be able to figure this out without so many comparisons, maybe an something along the lines of (enabled XOR flags) or something, would have to think about it...

Reply:
Who's gonna refresh their Karnaugh maps to figure this one out?

Reply:
I'll look it up and see what I can do

Reply:
You could use a 64 entry lookup table and find the answer with no branches.

Reply:
I pushed something that seemed to generate a small bit of assembly. Other thoughts welcome. https://godbolt.org/z/c777cWze6

Reply:
neat.  @beckerg , thoughts?

Reply:
Alex, maybe you could post your 64 entry lookup table idea so we could see the difference?

Reply:
Nice!  Now, which one is best?  other() ?  No jumps, but a judiciously placed likely/unlikely for the common case could change things...

Reply:
I was under the impression `once_again()` was the best version generated.

Reply:
it does look that way...

Reply:
I thought `call_can_compress` was best.  It has one branch, like `once_again` , and has a few more instructions, but they're all register operations while `once_again` has the load from memory.

Reply:
`call_can_compress` was the original method in this PR